domain/upgrade/state/state.go

```diff
@@ -329,3 +331,28 @@ AND db_completed_at IS NULL;`
 		return nil
 	}))
 }
+
+// SelectUpgradeInfo returns the upgrade info for the provided upgradeUUID
+func (st *State) SelectUpgradeInfo(ctx context.Context, upgradeUUID string) (upgrade.Info, error) {
+	db, err := st.DB()
+	if err != nil {
+		return upgrade.Info{}, errors.Trace(err)
+	}
+
+	getUpgradeQuery := "SELECT * AS &info.* FROM upgrade_info WHERE uuid = $M.info_uuid"
+	getUpgradeStmt, err := sqlair.Prepare(getUpgradeQuery, info{}, sqlair.M{})
+	if err != nil {
+		return upgrade.Info{}, errors.Annotatef(err, "preparing %q", getUpgradeQuery)
+	}
+
+	var upgradeInstance info
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, getUpgradeStmt, sqlair.M{"info_uuid": upgradeUUID}).Get(&upgradeInstance)
```

Comment:
`errors.Trace` just return `nil` for a `nil` error, so there's no need to check it.

You can just do:
```go
return errors.Trace(tx.Query(ctx, getUpgradeStmt, sqlair.M{"info_uuid": upgradeUUID}).Get(&upgradeInstance))
```